peps/pep-0800.rst

```diff
@@ -0,0 +1,352 @@
+PEP: 800
+Title: Solid bases in the type system
+Author: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 21-Jul-2025
+Python-Version: 3.15
+Post-History: `18-Jul-2025 <https://discuss.python.org/t/solid-bases-for-detecting-incompatible-base-classes/99280>`__
+
+
+Abstract
+========
+
+To analyze Python programs precisely, type checkers need to know when two classes can and cannot have a common child class.
+However, the information necessary to determine this is not currently part of the type system. This PEP adds a new
+decorator, ``@typing.solid_base``, that indicates that a class is a "solid base". Two classes that have distinct, unrelated
+solid bases cannot have a common child class.
+
+Motivation
+==========
+
+In type checking Python, an important concept is that of reachability. Python type checkers generally
+detect when a branch of code can never be reached, and and they warn users about such code. This is useful
+because unreachable code unnecessarily complicates the program, and its presence can be an indication of a bug.
+
+For example, in this program::
+
+    def f(x: bool) -> None:
+        if isinstance(x, str):
+            print("It's both!")
+
+both pyright and mypy (with ``--warn-unreachable``), two popular type checkers, will warn that the body of the
+``if`` block is unreachable, because if ``x`` is a ``bool``, it cannot also be a ``str``.
+
+Reachability is complicated in Python by the presence of multiple inheritance. If instead of ``bool`` and ``str``,
+we use two user-defined classes, mypy and pyright do not show any warnings::
+
+    class A: pass
+    class B: pass
+
+    def f(x: A):
+        if isinstance(x, B):
+            print("It's both!")
+
+This is correct, because there a class that inherits from both ``A`` and ``B`` could exist.
```

Comment:
"there a class" => "there is a class"

Reply:
I think it would make more sense here to say

> This is correct, because a class that inherits from both ``A`` and ``B`` could exist.